### DISCUSSION TITLE
Como configurar gitHook

### DISCUSSION THREAD

**Discussion by ThailanDev** [community/community#70578] [Programming Help]
Estou configurando em meu projeto a automação que verifica se o commit está seguindo a convenção de boas práticas.

Meu sistema operacional é o windows 10
Projeto feito em kotlin - android
Quando faço commit ele simplemente não roda os hook,como se eles não fosse executavel.

Procedimento:

1 - removi o sufixo sample do arquivo que fica dentro .git/hook
2 - dei todas as permições para arquivo ser lido e executavel.
3 - nome do arquivo "pre-commit" formato file

Resultado: Tentei rodar e não funcional [nenhuma mensagem de erro foi exibido] após isso, eu tentei.

4 - mudar o formato do arquivo para .sh

Nada aconteceu 😦

Lógica aplicada dentro do arquivo.

Verifico se a branch contem algumas das key word citadas.

#!/bin/bash BRANCH=$(git rev-parse --abbrev-ref HEAD) REGEX="^(feature|bugfix|release|hotfix)/[a-z0-9._-]+$"

if ! [[ $BRANCH =~ $REGEX ]]; then echo "Seu commit foi rejeitado pois não segue aos padrões definidos pelo time" echo "Por favor, altere o nome da sua branch de acordo com o padrão '(feature|bugfix|release|hotfix)/algumacoisa' " exit 1 fi

Alguém sabe o que devo está fazendo de errado?

**Comment by tuves** [comment]
Hi @ThailanDev, thanks for participating!
We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
